### PR TITLE
Add swift fibonacci search

### PR DESCRIPTION
**Pull request by erwolfe**
**Fixes issue:** #3054 

**Changes:**
1. Implementing fibonacci search in swift
2. Adding folder fibonacci_search in the search folder

### CODE REVIEW

File: code/search/fibonacci_search/fibonacci_search.swift [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        fibM = fibM2 + fibM1
+    }
+
+    // Set offset to -1
+    var offset = -1
+
+    // Continue looping until the index of element in the array is found
+    while(fibM > 1){
+        let i = minimumNumber(offset+fibM2, n-1)
+        if(numberArray[i] < x){
```
Comment (iattempt) [inline code/search/fibonacci_search/fibonacci_search.swift]: Parentheses around conditionals are not required and should be omitted.

File: code/search/fibonacci_search/fibonacci_search.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+ - parameters:
+     - numberArray: An array of ints that is being traversed to find an element.
+     - x: An int that is been looked for in the array.
+     - n: The size of the of the array.
+
+ - returns:
+ An int that is the index of the element in the array unless its not found and then -1 is returned.
+ */
+public func fibonacciSearch(_ numberArray: [Int],_ x: Int, _ n: Int) -> Int {
```
Comment (iattempt) [inline code/search/fibonacci_search/fibonacci_search.swift]: Although the coding style is not said put one space after each parameters, but I think you want put it...? so let it be rule. (don't forgot add this to coding style if you want.)

Comment (erwolfe) [reply code/search/fibonacci_search/fibonacci_search.swift]: So you want this?

```swift
public func fibonacciSearch(_ numberArray: [Int], _ x: Int, _ n: Int) -> Int {
```

File: code/search/fibonacci_search/fibonacci_search.swift (hunk 1 above)
Comment (erwolfe) [reply code/search/fibonacci_search/fibonacci_search.swift]: I'll change this right away!

File: code/search/fibonacci_search/fibonacci_search.swift (hunk 2 above)
Comment (iattempt) [reply code/search/fibonacci_search/fibonacci_search.swift]: yes.

**Comment by erwolfe** [comment]
I made both of the changes

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**